silicone/silicone.py

```diff
@@ -0,0 +1,620 @@
+    "s": "Statement/Subjective Statement",
+    "d": "Declarative Question",
+    "b": "Backchannel",
+    "f": '"Follow-me"',
+    "q": "Question",
+}
+
+IEMOCAP_E_DESCRIPTION = {
+    "ang": "Anger",
+    "dis": "Disgust",
+    "exc": "Excitement",
+    "fea": "Fear",
+    "fru": "Frustration",
+    "hap": "Happiness",
+    "neu": "Neutral",
+    "oth": "Other",
+    "sad": "Sadness",
+    "sur": "Surprise",
+    "xxx": "Unknown",
+}
+
+
+class SiliconeConfig(datasets.BuilderConfig):
+    """BuilderConfig for SILICONE."""
+
+    def __init__(
+        self,
+        text_features,
+        label_column,
+        data_url,
+        citation,
+        url,
+        label_classes=None,
+        **kwargs,
+    ):
+        """BuilderConfig for SILICONE.
+        Args:
+          text_features: `dict[string, string]`, map from the name of the feature
+            dict for each text field to the name of the column in the tsv file
+          label_column: `string`, name of the column in the csv/txt file corresponding
+            to the label
+          data_url: `string`, url to download the csv/text file from
+          citation: `string`, citation for the data set
+          url: `string`, url for information about the data set
+          label_classes: `list[string]`, the list of classes if the label is
+            categorical. If not provided, then the label will be of type
+            `datasets.Value('float32')`.
+          **kwargs: keyword arguments forwarded to super.
+        """
+        super(SiliconeConfig, self).__init__(version=datasets.Version("1.0.0", ""), **kwargs)
+        self.text_features = text_features
+        self.label_column = label_column
+        self.label_classes = label_classes
+        self.data_url = data_url
+        self.citation = citation
+        self.url = url
+
+class Silicone(datasets.GeneratorBasedBuilder):
+    """The Sequence labellIng evaLuatIon benChmark fOr spoken laNguagE (SILICONE) benchmark."""
+
+    BUILDER_CONFIGS = [
+        SiliconeConfig(
+            name="dyda_da",
+            description=textwrap.dedent(
+                """\
+            The DailyDialog Act Corpus contains multi-turn dialogues and is supposed to reflect daily
+            communication by covering topics about daily life. The dataset is manually labelled with
+             dialog act and emotions. It is the third biggest corpus of SILICONE with 102k utterances."""
+            ),
+            text_features={
+                "Utterance": "Utterance",
+                "Dialogue_Act": "Dialogue_Act",
+                "Dialogue_ID": "Dialogue_ID",
+            },
+            label_classes=["commissive", "directive", "inform", "question"],
+            label_column="Dialogue_Act",
+            data_url={
+                "train": os.path.join(_URL, "dyda", "train.csv"),
+                "dev": os.path.join(_URL, "dyda", "dev.csv"),
+                "test": os.path.join(_URL, "dyda", "test.csv"),
+            },
```

Comment:
```suggestion
            data_url={
                "train": _URL + "/dyda/train.csv",
                "dev": _URL + "/dyda/dev.csv",
                "test": _URL + "/dyda/test.csv"),
            },
```
Here we can avoid using `os.path.join` on urls. Indeed on Windows `os.path.join` groups the components using backslashes and the result may not be a valid url.

You can concatenate strings using `+` instead.

Can you also fix that for the other `data_url` dic defined for the other configuration ?